modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -127,3 +130,41 @@ func (Keeper) OnChanCloseConfirm(
 ) error {
 	return nil
 }
+
+// OnChanUpgradeTry performs the upgrade try step of the channel upgrade handshake.
+func (k Keeper) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, counterpartyVersion string) (string, error) {
+	if portID != icatypes.HostPortID {
+		return "", errorsmod.Wrapf(porttypes.ErrInvalidPort, "expected %s, got %s", icatypes.HostPortID, portID)
+	}
+
+	counterpartyVersion = strings.TrimSpace(counterpartyVersion)
+	if counterpartyVersion == "" {
```

Comment:
we can probably consolidate this

```suggestion
	if strings.TrimSpace(counterpartyVersion) == "" {
```